piccolo3/client/PiccoloCoolboxControl.py

```diff
@@ -71,16 +75,53 @@ async def get_current_temp(self):
     @property
     def target_temp(self):
         return self._target_temp
+
     @property
     def current_temp(self):
         return self._current_temp
 
+
+class PiccoloVoltage(PiccoloNamedClientComponent):
+    """manage voltage control on coolbox"""
+
+    NAME = "coolboxctrl"
+
+    def __init__(self, baseurl, name):
+
+        super().__init__(baseurl, name)
+
+        self._current_voltage = None
+
+        self._callbacks = []
+
+        self.add_task(self._get_current_voltage())
+
+    def register_callback(self, cb):
+        self._callbacks.append(cb)
+
+    async def _get_current_voltage(self):
+        u = 'current_voltage'
+        async for s in self.a_observe(u):
+            for cb in self._callbacks:
+                await cb(json.dumps((self.name, u, s)))
+            self._current_voltage = s
+
+    async def get_current_voltage(self):
+        self.log.info(self.uri('current_voltage'))
```

Comment:
ah, I suspect you copied something I put in there for my debugging benefit.

Reply:
Do you mean in terms of line 109 with the logging call?

Reply:
yes, just the logging. I think that was just for my benefit.

Reply:
Cool, I've removed.